temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -230,62 +302,36 @@ private Result handleWorkflowTaskWithEmbeddedQuery(
       }
       throw e;
     } finally {
-      if (stickyTaskQueueName == null && workflowRunTaskHandler != null) {
-        log.trace(
-            "Closing runner for {}-{} because of non-sticky worker configuration",
-            execution.getWorkflowId(),
-            execution.getRunId());
-        workflowRunTaskHandler.close();
-      } else {
+      if (useCache) {
         cache.markProcessingDone(execution);
+      } else if (workflowRunTaskHandler != null) {
+        workflowRunTaskHandler.close();
       }
     }
   }
 
-  private void resetStickyTaskQueue(WorkflowExecution execution) {
-    service
-        .futureStub()
-        .resetStickyTaskQueue(
-            ResetStickyTaskQueueRequest.newBuilder()
-                .setNamespace(namespace)
-                .setExecution(execution)
-                .build());
-  }
-
-  private Result handleQueryOnlyWorkflowTask(
-      PollWorkflowTaskQueueResponse.Builder workflowTask, Scope metricsScope) {
+  private Result createQueryResult(
+      PollWorkflowTaskQueueResponseOrBuilder workflowTask,
+      Optional<Payloads> queryResult,
+      Throwable e) {
     RespondQueryTaskCompletedRequest.Builder queryCompletedRequest =
         RespondQueryTaskCompletedRequest.newBuilder()
             .setTaskToken(workflowTask.getTaskToken())
             .setNamespace(namespace);
-    WorkflowExecution execution = workflowTask.getWorkflowExecution();
-    WorkflowRunTaskHandler workflowRunTaskHandler = null;
-    AtomicBoolean createdNew = new AtomicBoolean();
-    try {
-      workflowRunTaskHandler = getOrCreateWorkflowExecutor(workflowTask, metricsScope, createdNew);
-      Optional<Payloads> queryResult =
-          workflowRunTaskHandler.handleQueryWorkflowTask(workflowTask, workflowTask.getQuery());
-      if (stickyTaskQueueName != null && createdNew.get()) {
-        cache.addToCache(execution, workflowRunTaskHandler);
-      }
-      if (queryResult.isPresent()) {
-        queryCompletedRequest.setQueryResult(queryResult.get());
-      }
-      queryCompletedRequest.setCompletedType(QueryResultType.QUERY_RESULT_TYPE_ANSWERED);
-    } catch (Throwable e) {
+
+    if (e != null) {
+      queryCompletedRequest.setCompletedType(QueryResultType.QUERY_RESULT_TYPE_FAILED);
```

Comment:
Super nitpicky feel free to ignore: I would invert the conditional here. It's nice to read happy path first.